dbms/src/Common/ThreadMetricUtil.cpp

```diff
@@ -12,24 +12,24 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <Common/Stopwatch.h>
 #include <Common/ThreadMetricUtil.h>
 #include <Common/TiFlashMetrics.h>
 #include <common/types.h>
 
-std::atomic<UInt64> last_max_thds_metric_reset_ts{0};
-const std::chrono::duration<UInt64> max_thds_metric_reset_interval{60}; // 60s
+using Clock = std::chrono::steady_clock;
+using TimePoint = Clock::time_point;
+std::atomic<TimePoint> last_max_thds_metric_reset_ts{};
+const std::chrono::seconds max_thds_metric_reset_interval{60}; // 60s
 
 namespace DB
 {
 bool tryToResetMaxThreadsMetrics()
 {
-    using nanoseconds = std::chrono::duration<UInt64, std::nano>;
-    UInt64 last_max_thds_metric_reset_ts_tmp = last_max_thds_metric_reset_ts.load(std::memory_order_relaxed);
-    UInt64 now_ts = clock_gettime_ns_adjusted(last_max_thds_metric_reset_ts_tmp, CLOCK_MONOTONIC);
-    if (nanoseconds(now_ts) > nanoseconds(last_max_thds_metric_reset_ts_tmp) + max_thds_metric_reset_interval)
+    auto last_max_thds_metric_reset_ts_tmp = last_max_thds_metric_reset_ts.load(std::memory_order_relaxed);
```

Comment:
is overhead of new one same as clock_monotonic？

Reply:
actually, a little bit slower……

Reply:
then I think that will be bad，why bother changing into a slower one...

Reply:
> Class std::chrono::steady_clock represents a monotonic clock. The time points of this clock cannot decrease as physical time moves forward and the time between ticks of this clock is constant. This clock is not related to wall clock time (for example, it can be time since last reboot), **and is most suitable for measuring intervals**.

Reply:
it is better to use `std::chrono::steady_clock` to measure intervals.

Reply:
> actually, a little bit slower……

sorry for misleading, std::chrono::steady_clock is also CLOCK_MONOTONIC, so there is no difference in performance. Any other suggestions?

Reply:
Guess `std::chrono::steady_clock` actually calls `clock_gettime(MONOTONIC)`. you can try to confirm it.

Reply:
libstdc++: https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/src/c%2B%2B11/chrono.cc#L87

Reply:
libc++: https://github.com/llvm-mirror/libcxx/blob/master/src/chrono.cpp#L220

Reply:
please do things right.

Reply:
lgtm